### PR TITLE
Introduce ovirt-engine-repositories role

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by KKoukiou** [comment]
@mwperina  @machacekondra Just moved all files from other repo, and added the rpm related scripts (just copied the from vm-infra making changes where necessary)/

**Comment by KKoukiou** [comment]
Also can somebody with appropriate permissions enable the Travis CI for this repository?

**Comment by machacekondra** [comment]
ci build please

**Comment by machacekondra** [comment]
I've setup everything. Also in engine-setup repo. CI will start to work, once this patch is merged:

 https://gerrit.ovirt.org/#/c/85203/

**Comment by KKoukiou** [comment]
@machacekondra Super, thanks!

**Review by mwperina — APPROVED** [review]

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+VERSION="1.0.0"
```
Comment (machacekondra) [inline build.sh]: VERSION="1.1.0"

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+MILESTONE=
```
Comment (machacekondra) [inline build.sh]: MILESTONE=master

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+RPM_RELEASE="1"
```
Comment (machacekondra) [inline build.sh]: RPM_RELEASE="0.1.$MILESTONE.$(date -u +%Y%m%d%H%M%S)"

File: build.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+dist() {
+  echo "Creating tar archive '$TARBALL' ... "
+  sed \
+   -e "s|@RPM_VERSION@|$RPM_VERSION|g" \
+   -e "s|@RPM_RELEASE@|$RPM_RELEASE|g" \
+   -e "s|@PACKAGE_NAME@|$PACKAGE_NAME|g" \
+   -e "s|@PACKAGE_VERSION@|$PACKAGE_VERSION|g" \
+   < ovirt-ansible-repositories.spec.in > ovirt-ansible-repositories.spec
+
+  git ls-files | tar --files-from /proc/self/fd/0 -czf "$TARBALL" ovirt-ansible-vm-infra.spec
```
Comment (machacekondra) [inline build.sh]: git ls-files | tar --files-from /proc/self/fd/0 -czf "$TARBALL" ovirt-ansible-repositories.spec

File: ovirt-ansible-repositories.spec.in [hunk 5]
```diff
@@ -0,0 +1,54 @@
+%global rolename repositories
+%global roleprefix oVirt.
+%global roleprefix_legacy ovirt-
+%global ansible_roles_dir ansible/roles
+
+Name: @PACKAGE_NAME@
+Summary: Ansible role to create virtual machine infrastructure
```
Comment (machacekondra) [inline ovirt-ansible-repositories.spec.in]: Summary: Ansible role to setup oVirt/RHV repositories

File: ovirt-ansible-repositories.spec.in [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+Source0: http://resources.ovirt.org/pub/src/@PACKAGE_NAME@/@PACKAGE_NAME@-@PACKAGE_VERSION@.tar.gz
+License: ASL 2.0
+Group:          Virtualization/Management
+BuildArch:      noarch
+Url:            http://www.ovirt.org
+
+Requires: ansible >= 2.4
+
+%description
+This Ansible role is used to set the repositories required for
```
Comment (machacekondra) [inline ovirt-ansible-repositories.spec.in]: s/set/set up/

seems better for me, but I am not sure.

File: ovirt-ansible-repositories.spec.in [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+oVirt engine or host installation.
+
+%pretrans -p <lua>
```
Comment (machacekondra) [inline ovirt-ansible-repositories.spec.in]: You can remove this `%pretrans`, it was used only for roles which existed in 4.1

**Comment by machacekondra** [comment]
Thanks for contributing automation files 👍

File: build.sh [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+install() {
+  echo "Installing data..."
+  mkdir -p $PKG_DATA_DIR
+  mkdir -p $PKG_DOC_DIR
+
+  # Create a symlink, so legacy role name does work:
+  ln -f -s $PKG_DATA_DIR_ORIG $ROLENAME_LEGACY 
+  
+  cp -pR defaults/ $PKG_DATA_DIR
+  cp -pR library/ $PKG_DATA_DIR
```
Comment (machacekondra) [inline build.sh]: you can remove this line

**Comment by KKoukiou** [comment]
@mwperina @machacekondra Anything else I could change here? If not let's proceed with merging this one because tests for https://github.com/oVirt/ovirt-ansible-engine-setup/pull/1 are depending on this role.

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for an this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organisation's projects.

Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

**Comment by ovirt-infra** [comment]
Some tests failed

**Comment by machacekondra** [comment]
ci add to whitelist

**Comment by machacekondra** [comment]
ci build please

**Comment by ovirt-infra** [comment]
Some tests failed

**Comment by ovirt-infra** [comment]
Some tests failed

**Comment by machacekondra** [comment]
ci add to whitelist

**Comment by machacekondra** [comment]
ci build please

**Review by mwperina — APPROVED** [review]

**Comment by machacekondra** [comment]
@mwperina Do we want to create legacy symlink for newly created roles? Or are we ok with name just `oVirt.repositories`?

**Comment by mwperina** [comment]
No, new roles (unreleased in oVirt 4.1.z) should have only the new name, ie `oVirt.repositories`

File: ovirt-ansible-repositories.spec.in [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+BuildArch:      noarch
+Url:            http://www.ovirt.org
+
+Requires: ansible >= 2.4
+
+%description
+This Ansible role is used to set up the repositories required for
+oVirt engine or host installation.
+
+-- Remove the legacy directory before installing the symlink. This is known issue in RPM:
```
Comment (machacekondra) [inline ovirt-ansible-repositories.spec.in]: Please remove lines  22-28

File: build.sh [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+ROLE_NAME="oVirt.repositories"
+PACKAGE_NAME="ovirt-ansible-repositories"
+PREFIX=/usr/local
+DATAROOT_DIR=$PREFIX/share
+ROLES_DATAROOT_DIR=$DATAROOT_DIR/ansible/roles
+DOC_DIR=$DATAROOT_DIR/doc
+PKG_DATA_DIR=${PKG_DATA_DIR:-$ROLES_DATAROOT_DIR/$PACKAGE_NAME}
+PKG_DATA_DIR_ORIG=${PKG_DATA_DIR_ORIG:-$PKG_DATA_DIR}
+PKG_DOC_DIR=${PKG_DOC_DIR:-$DOC_DIR/$PACKAGE_NAME}
+ROLENAME_LEGACY="${ROLENAME_LEGACY:-$ROLES_DATAROOT_DIR/ovirt-repositories}"
```
Comment (machacekondra) [inline build.sh]: Please remove this line.

File: build.sh [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  echo "tar archive '$TARBALL' created."
+}
+
+install() {
+  echo "Installing data..."
+  mkdir -p $PKG_DATA_DIR
+  mkdir -p $PKG_DOC_DIR
+
+  # Create a symlink, so legacy role name does work:
+  ln -f -s $PKG_DATA_DIR_ORIG $ROLENAME_LEGACY
```
Comment (machacekondra) [inline build.sh]: Please remove line 43 and 44.

File: ovirt-ansible-repositories.spec.in [hunk 12]
```diff
@@ -0,0 +1,53 @@
+%global rolename repositories
+%global roleprefix oVirt.
+%global roleprefix_legacy ovirt-
```
Comment (machacekondra) [inline ovirt-ansible-repositories.spec.in]: Please remove this line.

File: ovirt-ansible-repositories.spec.in [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+%prep
+%setup -c -q
+
+%build
+
+%install
+export PKG_DATA_DIR_ORIG=%{_datadir}/%{ansible_roles_dir}/%{roleprefix}%{rolename}
+export PKG_DATA_DIR=%{buildroot}$PKG_DATA_DIR_ORIG
+export PKG_DOC_DIR=%{buildroot}%{_pkgdocdir}
+export ROLENAME_LEGACY=%{buildroot}%{_datadir}/%{ansible_roles_dir}/%{roleprefix_legacy}%{rolename}
```
Comment (machacekondra) [inline ovirt-ansible-repositories.spec.in]: Please remove this line.

File: ovirt-ansible-repositories.spec.in [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+sh build.sh install
+
+%files
+%{_datadir}/%{ansible_roles_dir}/%{roleprefix}%{rolename}
+%{_datadir}/%{ansible_roles_dir}/%{roleprefix_legacy}%{rolename}
```
Comment (machacekondra) [inline ovirt-ansible-repositories.spec.in]: Please remove this line.

**Comment by ifireball** [comment]
ci test please

**Review by machacekondra — APPROVED** [review]

**Comment by machacekondra** [comment]
ci buid please

**Comment by KKoukiou** [comment]
Thanks for quick reviews!

**Pull request merged**